app/models/manager_refresh/inventory_collection.rb

```diff
@@ -217,17 +217,20 @@ def fixed_attributes
       fixed_attributes
     end
 
+    # Returns all unique non saved fixed dependencies
     def fixed_dependencies
       fixed_attrs        = fixed_attributes
       fixed_dependencies = Set.new
       filtered_dependency_attributes.each do |key, value|
         fixed_dependencies += value if fixed_attrs.include?(key)
       end
-      fixed_dependencies
+
+      fixed_dependencies.select { |x| !x.saved? }
```

Comment:
might be nice to introduce `not_saved?` and `select(&:not_saved?)`

not sure

Also, `each_with_object` would clean this up as well.